core/src/packets/icmp/v4/mod.rs

```diff
@@ -419,11 +434,34 @@ mod tests {
         let packet = Mbuf::from_bytes(&ICMPV4_PACKET).unwrap();
```

Comment:
q: @drunkirishcoder for setting a good example (of how to add tests), do we want parse tests for each message type specifically, pulling in more byte arrays from one of these places: https://cloudshark.io/articles/how-to-get-sample-captures/? Know it's tedious, but sets a good example for adding more protocols, and you did something good by adding all these push_and_set* tests :).

Reply:
I don't think parsing each type is that interesting or error prone. and some types like time exceeded and packet too big require a large payload to be defined, not ideal as byte arrays.

Reply:
@drunkirishcoder yeah. that's fine. I guess more so for setting a good example for adding new types in the future (at least for ones more manageable than timeexceeded etc..., especially b/c many sites now have various packet examples. No biggie otherwise.